alacritty/src/event.rs

```diff
@@ -919,13 +919,16 @@ impl<'a, N: Notify + 'a, T: EventListener> ActionContext<'a, N, T> {
     fn update_cursor_blinking(&mut self) {
         // Get config cursor style.
         let mut cursor_style = self.config.terminal_config.cursor.style;
-        if self.terminal.mode().contains(TermMode::VI) {
+        let vi_mode = self.terminal.mode().contains(TermMode::VI);
+        if vi_mode {
             cursor_style = self.config.terminal_config.cursor.vi_mode_style.unwrap_or(cursor_style);
-        };
+        }
 
         // Check terminal cursor style.
         let terminal_blinking = self.terminal.cursor_style().blinking;
         let blinking = cursor_style.blinking_override().unwrap_or(terminal_blinking);
+        let blinking =
+            blinking && (vi_mode || self.terminal().mode().contains(TermMode::SHOW_CURSOR));
```

Comment:
```suggestion
        let blinking &= vi_mode || self.terminal().mode().contains(TermMode::SHOW_CURSOR);
```

Reply:
Should we really define this as two separate variables, one shadowing the other though? I feel like making the initial variable mutable is far superior. Even better might be just to add the `&& …` at the end of the definition.